packages/nuxt/src/pages/module.ts

```diff
@@ -186,6 +189,32 @@ export default defineNuxtModule({
       }
     })
 
+    const routeRulesMatcher = toRouteMatcher(
+      createRadixRouter({ routes: { ...nuxt.options.routeRules, ...nuxt.options.nitro?.routeRules } })
+    )
+    const getRouteRules = (path: string) =>
+      defu({}, ...routeRulesMatcher.matchAll(path).reverse()) as NitroRouteRules
+
+    nuxt.hook('build:manifest', async (manifest) => {
+      const pages = await resolvePagesRoutes()
+      await nuxt.callHook('pages:extend', pages)
+      const routes: string[] = []
+
+      for (const page of pages) {
+        const rules = getRouteRules(page.path)
+        if (rules.noScripts && page.file && !pages.some(p => p.file === page.file && p !== page)) {
+          routes.push(relative(nuxt.options.srcDir, page.file))
+        }
+      }
+
+      for (const file of routes) {
+        if (!manifest[file] || manifest[file].resourceType !== 'script') { continue }
+        if (routes.some(fileName => fileName in manifest)) {
+          manifest[file].file = ''
+        }
+      }
+    })
```

Comment:
What are you thinking about this section - do we still need it? I think we have two options:

1. `noScripts` means there are no scripts corresponding to given routes _at all_ - so navigating to them is a MPA navigation (hard reload). This is tricky as vite may well try to preload these (triggering chunk errors if we remove the files), and we probably have to write both a runtime manifest + vue router integration to achieve this. (Or maybe we can use a nitro transform plugin for the final build, which would also mean we can use rollup to tree-shake out other dependencies from the build.)
2. `noScripts` simply refers to the _initial_ load from the server

I think the simplest solution for now is to choose the latter option, at least to start with. But we should likely make this behaviour clear in the documentation, and maybe also mark this as experimental. (Maybe by naming it `experimentalNoScripts: true`?) What do you think @pi0 - is there a better way to mark a route rule experimental?

Reply:
Yes I think starting with the latter is a good start. Maybe should I have open a discussion about it before doing this 😅 ...
So client side navigation would still trigger a SPA behavior.
I guess we can remove this code ?

Reply:
`experimentalNoScripts` seems a nice idea!